### PR TITLE
Feature/simulator

### PR DESCRIPTION
**Pull request by konst-sauce**
Initial commit for 10 most common errors issue. 

Note: ObjC examples are still to be polished - they are on the way.

I've created a separate module for simulations - BacktraceSimulator. Also added a mock invisible XIB with primitive UI to test UI-related errors (like calling UI element on non-main thread or non-optional IBOutlet is nil (f.e. developer forgot to drag a reference in Interface Builder)).

Architecture is primitive, though flexible. Simulator cases are represented by a struct with case function as pointers and case title (to be shown in UI in a table view). We can add new simulation cases easily just by adding them to an array of available simulations (see BacktraceSimulator init()) or commenting existing cases to remove them.

Also merged some code a bit to reduce duplicated code - added to the new Common-Swift folder (same follows up for ObjC examples - will be in Common-ObjC folder (renamed from Common folder)).

Code review is not requested due to work in progress state.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,6 +6,8 @@
 build/
 DerivedData/
 
+Examples/Konstantin/
```
Comment (vlussenburg) [inline .gitignore:9]: Probably not needed :)

File: Examples/Common-ObjC/Keys.m [hunk 2]
```diff
@@ -5,7 +5,7 @@ @implementation Keys
 + (NSString const *) universeName {
     // Set your universe name here - https://support.backtrace.io/hc/en-us/articles/360040516451-What-is-a-submission-url-
     // e.g.: yourteam
-    return @"yourteam";
+    return @"konst-ryab";
```
Comment (vlussenburg) [inline Examples/Common-ObjC/Keys.m:8]: Would edit this out :)

File: Examples/Common-ObjC/Keys.m [hunk 3]
```diff
@@ -16,7 +16,7 @@ + (NSString const *) backtraceUrl {
 
 + (NSString const *) backtraceToken {
     // Set your submission token - https://support.backtrace.io/hc/en-us/articles/360040105172
-    return @"token";
+    return @"04aecb6b3da05e8d83f2a27f9b5f41352ac792428c5e5f6a94ff58d43ee14e46";
```
Comment (vlussenburg) [inline Examples/Common-ObjC/Keys.m:19]: Same here.

File: Examples/Common-Swift/BacktraceSimulator.swift [hunk 4]
```diff
@@ -0,0 +1,170 @@
+//
+//  BacktraceSimulator.swift
+//  Backtrace-iOS
+//
+
+import UIKit
+import Backtrace
+
+
```
Comment (vlussenburg) [inline Examples/Common-Swift/BacktraceSimulator.swift:9]: I think double empty lines create warnings in Xcode

File: Examples/Common-Swift/BacktraceSimulator.swift [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+    func caseTitle(atIndex index: Int) -> String?
+    func executeCase(atIndex index: Int) throws
+}
+
+
+class BacktraceSimulator: BacktraceSimulatorProtocol {
+    
+    // MARK: Error cases
+    
+    // 1. Non-optional is nil and we call some method
```
Comment (vlussenburg) [inline Examples/Common-Swift/BacktraceSimulator.swift:21]: Would add comments on top of the relevant function

File: Examples/Common-Swift/BacktraceSimulator.swift [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+    
+    // MARK: Variables
+    
+    fileprivate typealias CaseFunction = () -> ()
+    
+    private var availableCases: [ErrorCase] = []
+    private var mockView: MockView?
+    
+//    ProcessInfo
```
Comment (vlussenburg) [inline Examples/Common-Swift/BacktraceSimulator.swift:49]: Inconsistent `//` placement

File: Examples/Example-macOS-ObjC/ViewController.m [hunk 7]
```diff
@@ -12,6 +13,8 @@ @implementation ViewController
 - (void)viewDidLoad {
     [super viewDidLoad];
 
+//    BacktraceSimulator * simulator = [[BacktraceSimulator alloc] init];
```
Comment (vlussenburg) [inline Examples/Example-macOS-ObjC/ViewController.m:16]: Remove commented out code

File: .gitignore (hunk 1 above)
Comment (konst-sauce) [reply .gitignore:9]: I've created my own folder for some staff which shouldn't be pushed (like token and universe name) - but forgot to remove it in Keys class - so pushed anyway :-)

Comment (vlussenburg) [reply .gitignore:9]: Fair. Could you call it something that is not your name, like `LocalConfiguration` or something? Then we could all do it that way :) Not a biggie if not.

File: Examples/Example-iOS-PnP/ViewControllerPnP.swift [hunk 8]
```diff
@@ -134,43 +135,49 @@ extension ViewControllerPnP : AVPlayerViewControllerDelegate {
...
     }
 
 //    @available(iOS 8.0, *)
     func playerViewControllerDidStopPictureInPicture(_ playerViewController: AVPlayerViewController) {
-        
+        print("playerViewControllerDidStopPictureInPicture")
+//        present(playerVC!, animated: true) {}
     }
 
 //    @available(iOS 8.0, *)
```
Comment (vlussenburg) [inline Examples/Example-iOS-PnP/ViewControllerPnP.swift:163]: Why is this commented?

File: Examples/Example-iOS-PnP/ViewControllerPnP.swift [hunk 9]
```diff
@@ -41,8 +44,9 @@ class ViewControllerPnP : UIViewController {
                 .commit()
         }
         
+        avPlayer?.actionAtItemEnd = .pause
         playerVC?.player = avPlayer
-
+//        playerVC?.canStartPictureInPictureAutomaticallyFromInline = true
```
Comment (vlussenburg) [inline Examples/Example-iOS-PnP/ViewControllerPnP.swift:49]: why is this commented?
